### PR TITLE
Feature/Mac build

### PR DESCRIPTION
**Pull request by lebdron**
- Fix build and link of external shared libraries
- Fix CMake target link libraries
- Fix bad reference in yac gate impl
- Fix includes
- Fix type deduction in make_test_subscriber

### CODE REVIEW

**Comment by l4l** [comment]
Fixes build on archlinux :+1:

**Comment by neewy** [comment]
Also fixes protobuf 3.4.0 compatibility issues 👍

File: test/framework/test_subscriber.hpp [hunk 1]
```diff
@@ -148,9 +148,11 @@ namespace framework {
       std::unique_ptr<VerificationStrategy<T>> strategy_;
     };
 
-    template <template <typename K> class S, typename T, typename... Args>
+    template <template <typename K> class S,
```
Comment (grimadas) [inline test/framework/test_subscriber.hpp]: Add docs here pls

**Comment by grimadas** [comment]
Also add that protobuf, grpc  are downloaded by default

**Pull request merged**